src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -620,13 +620,17 @@ module.exports = function(
     } else {
       // If an update was already in progress, we should schedule an Update
       // effect even though we're bailing out, so that cWU/cDU are called.
-      if (typeof instance.componentDidUpdate === 'function') {
-        if (
-          oldProps !== current.memoizedProps ||
-          oldState !== current.memoizedState
-        ) {
-          workInProgress.effectTag |= Update;
-        }
+      if (
+        typeof instance.componentDidUpdate === 'function' &&
+        (oldProps !== current.memoizedProps ||
+          oldState !== current.memoizedState)
+      ) {
+        workInProgress.effectTag |= Update | Bailout;
```

Comment:
Ah. I see in this case you're actually adding both.

That means that if we're doing a simple two pass reconciliation where the second one is able to reuse the work from the first render, there will be no highlighting even though the component did in fact render?